project/src/main/webapp/gmail.js

```diff
@@ -15,191 +15,237 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-    const greetings =
-        ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-    // Pick a random greeting.
-    const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
-    // Add it to the page.
-    const greetingContainer = document.getElementById('greeting-container');
-    greetingContainer.innerText = greeting;
-}
+var authorizeButton = document.getElementById('authorize_button');
+var signoutButton = document.getElementById('signout_button');
 
-    // Array of API discovery doc URLs for APIs used by the quickstart
-    var DISCOVERY_DOCS = ["https://www.googleapis.com/discovery/v1/apis/gmail/v1/rest"];
+// Array of API discovery doc URLs for APIs used by the quickstart
+var DISCOVERY_DOCS = ["https://www.googleapis.com/discovery/v1/apis/gmail/v1/rest"];
 
-    // Authorization scopes required by the API; multiple scopes can be
-    // included, separated by spaces.
-    var SCOPES = 'https://www.googleapis.com/auth/gmail.readonly';
 
-    var authorizeButton = document.getElementById('authorize_button');
-    var signoutButton = document.getElementById('signout_button');
+// Authorization scopes required by the API; multiple scopes can be
+// included, separated by spaces.
+var SCOPES = 'https://www.googleapis.com/auth/gmail.readonly';
 
-    /**
-      *  On load, called to load the auth2 library and API client library.
-      */
-    function handleClientLoad() {
-    gapi.load('client:auth2', initClient);        
+
+/**
+  *  Sign out the user upon button click.
+  */
+function handleSignoutClick(event) {
```

Comment:
Are the changes to put the gmail auth code in a separate class in a different PR?

Reply:
different pr, this one was just to quickly push up a change that was breaking the main code

Reply:
the gmail auth in a separate class is in this pr: https://github.com/Enoch-Chou/capstoneproject/pull/12

Reply:
There's also changes here updating the callbacks to promises. I'll just review those unless they're meant to be in a different PR?